src/coffee/extend/ForceDirectedGraph.coffee

```diff
@@ -0,0 +1,154 @@
+Function::property = (prop, desc) ->
+  Object.defineProperty @prototype, prop, desc
+
+
+# Hacky Edge implementation
+class Edge
+  constructor: (source, target) ->
+    @source = source
+    @target = target
+
+  @property 'p1',
+    get: ->
+      return {
+        x: @target.x,
+        y: @target.y,
+        z: @target.z
+      }
+
+  @property 'x',
+    get: ->
+      return @source.x
+
+  @property 'y',
+    get: ->
+      return @source.y
+
+  @property 'z',
+    get: ->
+      return @source.z
+
+  clone: () ->
+    return new Edge(@source.clone(), @target.clone())
+
+
+class Vertex extends Vector
+  constructor: () ->
+    super
+    @displacement = new Vector()
+
+class Graph
+  constructor: () ->
+    @vertices = []
+    @edges = []
+
+  addVertex: (vertex) ->
+    @vertices.push(vertex)
+    return @
+
+  addEdge: (edge) ->
+    @edges.push(edge)
+    return @
+
+
+class ForceDirectedGraph extends Graph
```

Comment:
What do you think if we simply make this as `Graph`, and then rename the `generate()` to `forceDirected()` ?